src/charm.py

```diff
@@ -1924,6 +1920,10 @@ def _on_update_status(self, _) -> None:
         # Restart topology observer if it is gone
         self._observer.start_observer()
 
+        if self.unit.is_leader() and "refresh_remove_trigger" not in self.app_peer_data:
+            self.postgresql.drop_hba_triggers()
+            self.app_peer_data["refresh_remove_trigger"] = "True"
+
```

Comment:
Try to drop the trigger if the charm got updated. There's no post refresh hook ATM.